### PR TITLE
Update error500.php cannot use protected property

### PR DESCRIPTION
**Pull request by markusramsak**
cannot use protected property `params` -> use `->getBoundParams()`

### CODE REVIEW

**Comment by othercorey** [comment]
The annotation seems incorrect here if $error can be an instance of `Error`.

File: templates/Error/error500.php [hunk 1]
```diff
@@ -24,9 +24,9 @@
         <?= h($error->queryString) ?>
     </p>
 <?php endif; ?>
-<?php if (!empty($error->params)) : ?>
+<?php if (!empty($error->getBoundParams())) : ?>
```
Comment (othercorey) [inline templates/Error/error500.php:27]: If $error can be multiple classes, should probably wrap all of this in a instanceof check.

Comment (markusramsak) [reply templates/Error/error500.php:27]: but the Error class also only has getBoundParams() and protected params ...

Comment (markstory) [reply templates/Error/error500.php:27]: Which exception class has `getBoundParams()`? I only see it on `Database\Statement\Statement`.

Comment (ADmad) [reply templates/Error/error500.php:27]: This now needs to be a `method_exists()` check now or better yet an `instanceof` check for specific exception class. Suggested change will generate errors for other error classes.

**Pull request closed without merging**